client/src/placement-context.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useRef } from "react";
+import { useGleanClick } from "./telemetry/glean-context";
+import useSWR from "swr";
+import { PLACEMENT_ENABLED } from "./env";
+import { useUserData } from "./user-context";
+import { useLocation } from "react-router";
+
+export enum Status {
+  success = "success",
+  geoUnsupported = "geo_unsupported",
+  capReached = "cap_reached",
+}
+
+export interface Fallback {
+  click: string;
+  view: string;
+  copy: string;
+  image: string;
+  by: string;
+}
+
+export interface PlacementError {
+  status: Status.geoUnsupported | Status.capReached;
+}
+
+export interface PlacementStatus {
+  status: Status.success;
+  click: string;
+  view: string;
+  copy?: string;
+  image?: string;
+  fallback?: Fallback;
+}
+
+export type PlacementData = PlacementStatus | PlacementError;
+
+const PLACEMENT_TYPES = ["docs", "search"];
+
+function hasPlacement(pathname: string): boolean {
+  const [, , typ] = pathname.split("/");
+  return PLACEMENT_TYPES.includes(typ?.toLowerCase?.());
+}
```

Comment:
Not sure if I'm happy with that. But I couldn't see a better way. Eventually we should have a single request in this central place, but it depends on the page type with what parameters we query the api.

Reply:
I would tend to prefer a RegExp here: `/\/[^\/]+\/(docs|search)\//i`